api4/file.go

```diff
@@ -603,9 +603,19 @@ func getFilePreview(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	fileReader, err := c.App.FileReader(info.PreviewPath)
 	if err != nil {
-		c.Err = err
-		c.Err.StatusCode = http.StatusNotFound
-		return
+		if info.PreviewPath != "" && *c.App.Config().ServiceSettings.EnableOfficeFilePreviews && info.IsUnoconvSupported() {
```

Comment:
This tries to generate the preview in first render (this is because you can enable/disable this feature, so some previews can be not generated in the first upload).